### DISCUSSION TITLE
Curobo errors encountered during installation

### DISCUSSION THREAD

**Discussion by whwdadada** [NVlabs/curobo#59] [Software Q&A] [answered]
the notification is"ValueError: mutable default <class 'curobo.wrap.reacher.evaluator.TrajEvaluatorConfig'> for field traj_evaluator_config is not allowed: use default_factory" . 
I am very confused with it because i have corrected it according to some advice provided by ChatGPT
The original code is :
@dataclass
class TrajEvaluatorConfig:
    max_acc: float = 15.0
    max_jerk: float = 500.0
    cost_weight: float = 0.01
    min_dt: float = 0.001
    max_dt: float = 0.1

class TrajEvaluator(TrajEvaluatorConfig):
    def __init__(self, config: Optional[TrajEvaluatorConfig] = None):
        if config is None:
            config = TrajEvaluatorConfig()
        super().__init__(**vars(config))

And the corrected code is:
@dataclass
class TrajEvaluatorConfig:
    max_acc: float = 15.0
    max_jerk: float = 500.0
    cost_weight: float = 0.01
    min_dt: float = 0.001
    max_dt: float = 0.1

@dataclass
class TrajEvaluator:
    traj_evaluator_config: TrajEvaluatorConfig = field(default_factory=TrajEvaluatorConfig)
    
    def __post_init__(self):
        # 手动处理可变默认值
        if not isinstance(self.traj_evaluator_config, TrajEvaluatorConfig):
            self.traj_evaluator_config = TrajEvaluatorConfig(**vars(self.traj_evaluator_config))
    
    def __init__(self, config: Optional[TrajEvaluatorConfig] = None):
        if config is not None:
            self.traj_evaluator_config = config

Are there people who know the reason this error still occurs

**Comment by balakumar-s** [comment]
What python version are you trying to install in?

**Comment by whwdadada** [comment, marked as answer]
Thank you for your response. I have solved this problem and I find this reason is that I use the newest version of the anaconda. Its version of python used is 3.11. In fact, the version 3.8 of python is suitable for installation. Therefore, will it to be more notable if there is a notification in the page of installation?
Thanks again.









 王浩文
***@***.***











Original:
From：Balakumar Sundaralingam ***@***.***>Date：2023-11-29 23:45:51(中国 (GMT+08:00))To：NVlabs/curobo ***@***.***>Cc：Haowen Wang ***@***.***> , Author ***@***.***>Subject：Re: [NVlabs/curobo] Curobo errors encountered during installation (Discussion #59)
What python version are you trying to install in?

**Reply by balakumar-s** [reply]
Thanks for letting us know. I will add a hint that it may not work on 3.11. We have it running on 3.7, 3.8, and 3.10.
